http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcSecurityRealmTest.java

```diff
@@ -0,0 +1,187 @@
+        clientConfiguration.setUseResourceRoleMappings(true);
+        final JwtClaims jwtClaims = new JwtClaims();
+        final Map<String, Object> resourceAccess = new HashMap<>();
+        resourceAccess.put("SomeResource", createRoles("roleA"));
+        resourceAccess.put("SpecialResource", createRoles("roleB", "roleC"));
+        jwtClaims.setClaim("resource_access", resourceAccess);
+        jwtClaims.setClaim("realm_access", createRoles("roleC", "roleD"));
+
+        RefreshableOidcSecurityContext securityContext = new RefreshableOidcSecurityContext(clientConfiguration,
+                null, null, new AccessToken(jwtClaims), null, null, null);
+        OidcPrincipal principal = new OidcPrincipal("john", securityContext);
+
+        // test
+        RealmIdentity identity = realm.getRealmIdentity(principal);
+        AuthorizationIdentity authorizationIdentity = identity.getAuthorizationIdentity();
+        final Attributes.Entry roles = authorizationIdentity.getAttributes().get(KEY_ROLES);
+
+        // verification
+        assertEquals(3, roles.size());
+        assertTrue(roles.contains("roleB"));
+        assertTrue(roles.contains("roleC"));
+        assertTrue(roles.contains("roleD"));
+    }
+
+    @Test
+    public void testGetRealmIdentityOnlyRealmRoles() throws RealmUnavailableException {
+        // setup
+        final OidcClientConfiguration clientConfiguration = new OidcClientConfiguration();
+        clientConfiguration.setResource("SpecialResource");
+        clientConfiguration.setUseRealmRoleMappings(true);
+        final JwtClaims jwtClaims = new JwtClaims();
+        final Map<String, Object> resourceAccess = new HashMap<>();
+        resourceAccess.put("SpecialResource", createRoles("roleB", "roleC"));
+        jwtClaims.setClaim("resource_access", resourceAccess);
+        jwtClaims.setClaim("realm_access", createRoles("roleC", "roleD"));
+
+        RefreshableOidcSecurityContext securityContext = new RefreshableOidcSecurityContext(clientConfiguration,
+                null, null, new AccessToken(jwtClaims), null, null, null);
+        OidcPrincipal principal = new OidcPrincipal("john", securityContext);
+
+        // test
+        AuthorizationIdentity authorizationIdentity = realm.getRealmIdentity(principal).getAuthorizationIdentity();
+        final Attributes.Entry roles = authorizationIdentity.getAttributes().get(KEY_ROLES);
+
+        // verification
+        assertEquals(2, roles.size());
+        assertTrue(roles.contains("roleC"));
+        assertTrue(roles.contains("roleD"));
+    }
+
+    @Test
+    public void testGetRealmIdentityOnlyResourceRoles() throws RealmUnavailableException {
+        // setup
+        final OidcClientConfiguration clientConfiguration = new OidcClientConfiguration();
+        clientConfiguration.setResource("SpecialResource");
+        clientConfiguration.setUseRealmRoleMappings(false);
+        clientConfiguration.setUseResourceRoleMappings(true);
```

Comment:
It would be great to also add a test for the case where they are both set to `false`.